daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala

```diff
@@ -438,3 +389,100 @@ object Value extends CidContainer1WithDefaultCidResolver[Value] {
   val ValueNil: ValueList[Nothing] = ValueList(FrontStack.empty)
   val ValueNone: ValueOptional[Nothing] = ValueOptional(None)
 }
+
+sealed abstract class ValueInstances {
+  // Order of GenMap entries is relevant for this equality.
+  implicit def `Value Equal instance`[Cid: Equal]: Equal[Value[Cid]] =
+    new `Value Equal instance`[Cid] {
+      override val E = Equal[Cid]
+    }
+}
+
+private final class `Value Order instance`[Cid](implicit val E: Order[Cid])
```

Comment:
I am a bit concern we have an order on `Value`. 
It will be very tricky to keep in think with the ordering of  `SValue` `com.digitalasset.daml.lf.speedy.svalue.Ordering`

Reply:
Yes, it will, but it is accompanied by a reasonable scalacheck-based testing tactic, to be implemented. Anyway, we signed up for this by having `ValueGenMap` accept wildly wrong input.

Reply:
Maybe we could generalize `com.digitalasset.daml.lf.speedy.svalue.OrderingSpec` to test both implementation.

Reply:
@remyhaemmerle-da This is what I have in mind 93decad0ffb8c43d70181d401cbfb8f5340de71f , I just need to convert Value to SValue to do a similar check for those.